src/devtools/client/debugger/src/actions/preview.js

```diff
@@ -73,12 +73,14 @@ export function setPreview(cx, expression, location, tokenPos, cursorPos, target
       result = createPrimitiveValueFront(undefined);
     }
 
-    const root = new ValueItem({
+    let root = new ValueItem({
       name: expression,
       path: expression,
       contents: result,
     });
     await root.loadChildren();
+    // recreate the ValueItem to update its loading state
+    root = root.recreate();
```

Comment:
This interaction sequence:
1. instantiate
2. load
3. force-recreate

Feels unfortunately complex.

Reply:
Yes it does. `ValueItem` was designed to be used by the `ObjectInspector` internally and here we're kind of abusing it...